actions/bump-and-release-cloud-plugin/action.yaml

```diff
@@ -0,0 +1,308 @@
+name: Bump and Release cloud plugin
+description: |
+  Bumps the version of a cloud plugin and tags the release. This action only builds Go binaries for the linux platform
+  as that is the only platform that Grafana Cloud uses.
+
+inputs:
+  default-branch:
+    description: |
+      The default branch of the repository. This is used to checkout the repository.
+    default: main
+    required: true
+  github-token:
+    description: |
+      A GitHub token used to commit the version bump and create a tag. This will need to have write access to the
+      default branch of the repository.
+    required: true
+  gcs-bucket:
+    description: GCS bucket to upload too
+    required: true
+  gcs-service-account-creds:
+    description: Credentials JSON to log into GCP and upload plugin to bucket
+    required: true
+  package-manager:
+    description: |
+      The package manager to use for building the plugin. This can be either `yarn` or `npm`.
+    default: yarn
+    required: true
+  release-level:
+    description: |
+      The level of the release. This can be one of `major`, `minor`, `patch`, or `prerelease`.
+      See https://semver.org/ for more information.
+    required: true
+  signing-token:
+    description: |
+      A Grafana access policy token to sign the plugin with.
+      See https://grafana.com/developers/plugin-tools/publish-a-plugin/sign-a-plugin#generate-an-access-policy-token
+    required: true
+  update-changelog:
+    description: |
+      Whether to update the changelog with the new version. When updating will replace the `## Unreleased` section
+      with `## Unreleased\n\n## v*.*.*`. Giving the top of the changelog a new unreleased section.
+    default: 'false'
+    required: true
+outputs:
+  version:
+    description: The semantic version that the package.json was bumped too
+    value: ${{ steps.output-version.outputs.version }}
+  version-tag:
+    description: The tag that was created for the version
+    value: ${{ steps.output-version.outputs.version-tag }}
+
+runs:
+  using: composite
+  steps:
+    - name: Specify GOPRIVATE env
+      shell: bash
+      run: |
+        echo "GOPRIVATE=github.com/grafana/*" >> $GITHUB_ENV
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        ref: ${{ inputs.default-branch }}
+        token: ${{ inputs.github-token }}
+        fetch-depth: 0
+
+    - name: Setup credentials to access Grafana private repositories
```

Comment:
I wonder if this should be an optional step, considering shared-workflows is a public repository, isn't there some parametrization that could be done here to allow people who don't have access to Grafana private repositories to use this to release their plugins?

In case cloud plugins are internal only and different from other plugins this makes sense, in which case this action is expected to not work for members without access to Grafana private repositories.

Should we leave a note on the readme to make this clear if that's the case?

Reply:
Good point, I think we could modify L#58 to instead of `GOPRIVATE=github.com/grafana/*` to be `GOPRIVATE=github.com/${{ github.repo.org }}/*` (might have my context variable wrong will double check) So that it's org agnostic. 

And then update the name on this step to just `Setup credentials to access private repositories` since it's really just slapping the token before github.com.

Reply:
> In case cloud plugins are internal only and different from other plugins this makes sense, in which case this action is expected to not work for members without access to Grafana private repositories.

Not necessarily, but so far to my knowledge cloud plugins have been internal provisioned plugins. That doesn't mean this couldn't build a OSS plugin, right now it just builds the linux binaries and not all the other possible binaries if there is a go backend.